app/src/main/java/org/openedx/app/MainViewModel.kt

```diff
@@ -1,20 +1,58 @@
 package org.openedx.app
 
+import androidx.lifecycle.LifecycleOwner
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.launch
 import org.openedx.core.BaseViewModel
 import org.openedx.core.config.Config
+import org.openedx.course.domain.interactor.CourseInteractor
+import org.openedx.dashboard.notifier.DashboardEvent
+import org.openedx.dashboard.notifier.DashboardNotifier
 
 class MainViewModel(
-    private val config: Config
+    private val config: Config,
+    private val notifier: DashboardNotifier,
+    private val interactor: CourseInteractor
 ) : BaseViewModel() {
 
     private val _isBottomBarEnabled = MutableLiveData(true)
     val isBottomBarEnabled: LiveData<Boolean>
         get() = _isBottomBarEnabled
 
+    private val _navigateToDiscovery = MutableSharedFlow<Boolean>()
+    val navigateToDiscovery: SharedFlow<Boolean>
+        get() = _navigateToDiscovery.asSharedFlow()
+
     val isDiscoveryTypeWebView get() = config.getDiscoveryConfig().isViewTypeWebView()
 
+    val isProgramTypeWebView get() = config.getProgramConfig().isViewTypeWebView()
+
+    override fun onCreate(owner: LifecycleOwner) {
+        super.onCreate(owner)
+        viewModelScope.launch {
+            notifier.notifier.collect {
+                if (it is DashboardEvent.NavigationToDiscovery) {
+                    _navigateToDiscovery.emit(true)
+                } else if (it is DashboardEvent.CourseEnrolled && it.courseId.isNotEmpty()) {
+                    try {
+                        interactor.enrollInACourse(it.courseId)
+                        notifier.send(DashboardEvent.CourseEnrolledSuccess(it.courseId))
+                    } catch (
+                        e: Exception
+                    ) {
+                        e.printStackTrace()
+                        notifier.send(DashboardEvent.CourseEnrolledError(e))
+                    }
+                }
```

Comment:
It's a nice workaround to not connect the course module, but controlling it via the event bus is non-obvious and a bit confusing. We can do it in a cleaner way.
I would recommend creating an interface like `EnrollInACourseUseCase` or `EnrollInACourseInteractor` in the `core` module and implementing that interface in the `CourseInteractor`.
After that, all you need to do is simply provide/inject the `EnrollInACourseInteractor` interface into `ProgramViewModel`.

Reply:
updated